src/libs/OptionsListUtils.ts

```diff
@@ -2316,12 +2333,39 @@ function filterOptions(options: Options, searchInputValue: string): Options {
         };
     }, options);
 
-    const recentReports = matchResults.recentReports.concat(matchResults.personalDetails);
+    let {recentReports, personalDetails} = matchResults;
+
+    if (sortByReportTypeInSearch) {
+        recentReports = recentReports.concat(matchResults.personalDetails);
+        personalDetails = [];
+        recentReports = orderOptions(recentReports, searchValue);
+    }
+
+    let userToInvite = null;
+    if (canInviteUser) {
+        const noOptions = recentReports.length + personalDetails.length === 0;
+        const noOptionsMatchExactly = !personalDetails
+            .concat(recentReports)
+            .find((option) => option.login === PhoneNumber.addSMSDomainIfPhoneNumber(searchValue ?? '').toLowerCase() || option.login === searchValue?.toLowerCase());
+        if (
+            searchValue &&
+            (noOptions || noOptionsMatchExactly) &&
+            !isCurrentUser({login: searchValue} as PersonalDetails) &&
+            selectedOptions.every((option) => 'login' in option && option.login !== searchValue) &&
+            ((Str.isValidEmail(searchValue) && !Str.isDomainEmail(searchValue) && !Str.endsWith(searchValue, CONST.SMS.DOMAIN)) ||
+                (parsedPhoneNumber.possible && Str.isValidE164Phone(LoginUtils.getPhoneNumberWithoutSpecialChars(parsedPhoneNumber.number?.input ?? '')))) &&
+            !excludeLogins.find((optionToExclude) => optionToExclude === PhoneNumber.addSMSDomainIfPhoneNumber(searchValue).toLowerCase()) &&
+            (searchValue !== CONST.EMAIL.CHRONOS || Permissions.canUseChronos(betas)) &&
+            !excludeUnknownUsers
```

Comment:
How about creating a function for [these](https://github.com/Expensify/App/pull/40235/files#diff-7d43f3c3e4decdb0057928e895f456d51c0f4081837edd5e2c4ed7ddd570e274R2351-R2359) if conditions part as the same set of conditions is used twice in the file